bwapi/bwdata.py

```diff
@@ -524,7 +524,7 @@ def get_date_range_comparison(
         Raises:
             KeyError:       If you fail to pass in a date range
         """
-        query_id = self.ids[name]
+        query_id = self.get_resource_id(name)
```

Comment:
NABD but the word "resource" has special semantics in BW. I'm not sure to what degree that is reflected in the API, but I'd suggest avoiding it to be on the safe side

Reply:
After a chat, decided to keep the resource name internally consistent with other parts of the SDK (e.g. BWResources), while recognising that "resource" has a different meaning elsewhere